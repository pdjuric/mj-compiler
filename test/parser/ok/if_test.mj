program if_test

int testCaseNo;
{
    bool T(int t) {
        print(t, 2);
        return true;
    }

    bool F(int t) {
        print(t, 2);
        return false;
    }

    void inThen() {
        print('t',2);
        print('h',1);
        print('e',1);
        print('n',1);
    }

    void next() {
        print(eol);
        print('t',1);
        print('e',1);
        print('s',1);
        print('t',1);
        print(testCaseNo, 4);
        print(':', 1);
        print(' ', 1);
        testCaseNo++;
    }

    void main() {
        testCaseNo = 0;

        next(); 		if (T(0)) inThen();
        next();		if (F(0)) inThen(); // 492
        next();		if (T(0) && T(1)) inThen();
        next();		if (F(0) && T(1)) inThen();
        next();		if (T(0) && F(1)) inThen();
        next();		if (F(0) && F(1)) inThen();
        next();		if (T(0) || T(1)) inThen();
        next();		if (F(0) || T(1)) inThen();
        next();		if (T(0) || F(1)) inThen();

        next();		if (T(0) && T(1) || T(2)) inThen();
        next();		if (T(0) && T(1) || F(2)) inThen();
        next();		if (T(0) && F(1) || T(2)) inThen();
        next();		if (T(0) && F(1) || F(2)) inThen();
        next();		if (F(0) && T(1) || T(2)) inThen();
        next();		if (F(0) && T(1) || F(2)) inThen();
        next();		if (F(0) && F(1) || T(2)) inThen();
        next();		if (F(0) && F(1) || F(2)) inThen();
        next();		if (T(0) || T(1) && T(2)) inThen();
        next();		if (T(0) || T(1) && F(2)) inThen();
        next();		if (T(0) || F(1) && T(2)) inThen();
        next();		if (T(0) || F(1) && F(2)) inThen();
        next();		if (F(0) || T(1) && T(2)) inThen();
        next();		if (F(0) || T(1) && F(2)) inThen();
        next();		if (F(0) || F(1) && T(2)) inThen();
        next();		if (F(0) || F(1) && F(2)) inThen();
        next();		if (T(0) || T(1) || T(2)) inThen();
        next();		if (T(0) || T(1) || F(2)) inThen();
        next();		if (T(0) || F(1) || T(2)) inThen();
        next();		if (T(0) || F(1) || F(2)) inThen();
        next();		if (F(0) || T(1) || T(2)) inThen();
        next();		if (F(0) || T(1) || F(2)) inThen();
        next();		if (F(0) || F(1) || T(2)) inThen();
        next();		if (F(0) || F(1) || F(2)) inThen();
        next();		if (T(0) && T(1) && T(2)) inThen();
        next();		if (T(0) && T(1) && F(2)) inThen();
        next();		if (T(0) && F(1) && T(2)) inThen();
        next();		if (T(0) && F(1) && F(2)) inThen();
        next();		if (F(0) && T(1) && T(2)) inThen();
        next();		if (F(0) && T(1) && F(2)) inThen();
        next();		if (F(0) && F(1) && T(2)) inThen();
        next();		if (F(0) && F(1) && F(2)) inThen();



    }

}
