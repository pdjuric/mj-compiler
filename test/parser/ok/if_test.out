
test   0:  0 then
test   1:  0
test   2:  0 1 then
test   3:  0
test   4:  0 1
test   5:  0
test   6:  0 then
test   7:  0 1 then
test   8:  0 then
test   9:  0 1 then
test  10:  0 1 then
test  11:  0 1 2 then
test  12:  0 1 2
test  13:  0 2 then
test  14:  0 2
test  15:  0 2 then
test  16:  0 2
test  17:  0 then
test  18:  0 then
test  19:  0 then
test  20:  0 then
test  21:  0 1 2 then
test  22:  0 1 2
test  23:  0 1
test  24:  0 1
test  25:  0 then
test  26:  0 then
test  27:  0 then
test  28:  0 then
test  29:  0 1 then
test  30:  0 1 then
test  31:  0 1 2 then
test  32:  0 1 2
test  33:  0 1 2 then
test  34:  0 1 2
test  35:  0 1
test  36:  0 1
test  37:  0
test  38:  0
test  39:  0
test  40:  0